index.bs

```diff
@@ -635,7 +641,20 @@ All fields following the sample entry 4CC are expressed as double digit decimals
 
 <assert>The chromaSubsampling parameter value, represented by a three-digit decimal, SHALL have its first digit equal to subsampling_x and its second digit equal to subsampling_y</assert>. <assert>If both subsampling_x and subsampling_y are set to 1, then the third digit SHALL be equal to chroma_sample_position, otherwise it SHALL be set to 0</assert>.
 
-<assert>The colorPrimaries, transferCharacteristics, matrixCoefficients, and videoFullRangeFlag parameter values SHALL equal the value of matching fields in the [=Sequence Header OBU=], if color_description_present_flag is set to 1</assert>, <assert>otherwise they SHOULD not be set, defaulting to the values below</assert>. The videoFullRangeFlag is represented by a single digit.
+The colorPrimaries, transferCharacteristics, matrixCoefficients, and videoFullRangeFlag parameter values are set as follows:
+- <assert>If a 'colr' box with colour_type set to 'nclx' is present, the colorPrimaries, transferCharacteristics, matrixCoefficients, and videoFullRangeFlag parameter values SHALL equal the values of matching fields in the 'colr' box</assert>.
+- Otherwise, a 'colr' box with colour_type set to 'nclx' is absent.
+    - <assert>The videoFullRangeFlag parameter value SHALL equal the color_range flag in the [=Sequence Header OBU=]</assert>.
+    - <assert>If the color_description_present_flag is set to 1 in the [=Sequence Header OBU=], the colorPrimaries, transferCharacteristics, and matrixCoefficients parameter values SHALL equal the values of matching fields in the [=Sequence Header OBU=]</assert>.
+    - Otherwise, the color_description_present_flag is set to 0 in the [=Sequence Header OBU=]. <assert>The colorPrimaries, transferCharacteristics, and matrixCoefficients parameter values SHOULD be set to the default values below</assert>.
+
+NOTE: It is encouraged that a 'colr' box be present. The 'colr' box simplifies the task of the codecs parameter generator.
+
+NOTE: If the encoded video uses a color space that is not registered in [[CICP]] and the file doesn't explicitly code 2/2/2 either in the 'colr' box or in the [=Sequence Header OBU=] (with color_description_present_flag set to 1), the derived codecs parameter string will declare ITU-R BT.709 (1/1/1). To avoid this behavior, a 'colr' box should be used.
+
+The videoFullRangeFlag is represented by a single digit.
+
+<assert>If the codecs parameter string ends with ".0.110.01.01.01.0" (containing all the default values below), that trailing part of the string SHOULD be omitted</assert>.
```

Comment:
Maybe this could be changed from SHOULD to MAY? I think it would be an implementers choice to include it or not.

Reply:
Thanks for the comment. It's a SHOULD rather than a MAY because it is inefficient to include all the default values. But in the end, it's an implementer's choice. The SHOULD just gives more indication as to what needs to be done than a MAY. Unless we have a use case that needs to include the default values, I think we should keep the SHOULD.